### PR TITLE
Makes it possible to set the initial camera position for the `Model3D` component as a tuple of (alpha, beta, radius)

### PR DESCRIPTION
**Pull request by mbahri**
## Description

This change addresses improvement request https://github.com/gradio-app/gradio/issues/5167 and makes it possible to set the initial ArcCamera position for Model3D (input and output).

Closes: #5167

### CODE REVIEW

**Comment by abidlabs** [comment]
Thanks @mbahri for the contribution! Instead of adding 3 separate parameters, could we add a single parameter `initial_position` which takes in a `tuple` of `(alpha, beta, radius)`?

**Comment by mbahri** [comment]
> Thanks @mbahri for the contribution! Instead of adding 3 separate parameters, could we add a single parameter `initial_position` which takes in a `tuple` of `(alpha, beta, radius)`?

I'll have a look :)

**Comment by abidlabs** [comment]
Did you get a chance to look into this @mbahri?

**Comment by mbahri** [comment]
Hi @abidlabs, yes I've made the change in the code, just haven't had time to test it yet. I'll update the PR today

**Comment by abidlabs** [comment]
Sounds good thanks @mbahri

**Comment by abidlabs** [comment]
Thanks @mbahri for pushing the changes. Some tests may need to be fixed, I can take care of that.

**Comment by abidlabs** [comment]
whoops I think we might have conflicted right there

**Comment by abidlabs** [comment]
Are you still making changes to the PR or should I take over?

**Comment by mbahri** [comment]
Hey @abidlabs, sorry, I didn't expect you to be working on this now. I've made some minor formatting changes and added a docstring for the new parameter.

I'm not too familiar with your coding style so please don't hesitate to rename variables or the type alias `NumberOrNone = int | float | None` I added for brevity to the model3d.py file (L.22).

**Comment by abidlabs** [comment]
Made a few tweaks -- for example the conversion factor from radians to degrees should be pi/180, not pi/360, correct?

**Comment by mbahri** [comment]
Correct, I mixed it up because I needed to set the angle to pi/2 to have my test models visually do a "180" on screen

I'd like to change the docstring for formatting and specifying which unit the angles are in but don't want to interfere now:
```
initial_camera_position: Tuple `(alpha, beta, radius)`. Sets the starting longitudinal rotation (`alpha`), latitudinal rotation (`beta`) and distance to the 3D model being displayed (`radius`) for the camera. Set any value to None to use the default value. `alpha` and `beta` should be provided in degrees.
```

**Comment by abidlabs** [comment]
Does this work? 

>  camera_position: initial camera position of scene, provided as a tuple of (alpha, beta, radius). Each value is optional. If provided, alpha and beta should be in degrees from -180 to 180 reflecting the angular position along the longitudinal and latitudinal axes, respectively. Radius corresponds to the distance from the center of the object to the camera.

**Comment by mbahri** [comment]
Sure, I just changed my commit to add backticks for the doc

File: js/model3D/interactive/Model3DUpload.svelte [hunk 1]
```diff
@@ -101,6 +108,19 @@
 			scene,
 			() => {
 				scene.createDefaultCamera(true, true, true);
+				var helperCamera = scene.activeCamera;
```
Comment (dawoodkhan82) [inline js/model3D/interactive/Model3DUpload.svelte:111]: Seems like we need a null check on `helperCamera` for the typescript errors.

**Comment by mbahri** [comment]
> from -180 to 180

Do we need to specify this range? I'm checking a couple things rn

File: js/model3D/interactive/Model3DUpload.svelte (hunk 1 above)
Comment (mbahri) [reply js/model3D/interactive/Model3DUpload.svelte:111]: The current code assumes `scene.createDefaultCamera` succeeds, do you want to add error handling (and in that case, what needs to happen?) or just assume there's an active camera?

I'm pushing a change that keeps assuming the call worked - as a starting point

**Comment by abidlabs** [comment]
> Do we need to specify this range? I'm checking a couple things rn

ah maybe not, I thought the range in babylon was -pi to pi radians, but I haven't tested to see if the values wrap around.

**Comment by abidlabs** [comment]
Used this opportunity to do some further refactoring & reduce the duplicated code in the component (not strictly related to your changes @mbahri)

File: js/model3D/interactive/Model3DUpload.svelte (hunk 1 above)
Comment (dawoodkhan82) [reply js/model3D/interactive/Model3DUpload.svelte]: Actually looks like the type error is gone, so should be fine as is.

**Comment by dawoodkhan82** [comment]
Code looks good to me!

**Review by dawoodkhan82 — APPROVED** [review]

**Comment by abidlabs** [comment]
Cool thanks! And thanks @mbahri for contributing this feature & bearing with us as we reviewed it!

**Pull request merged**